Mathlib/Combinatorics/Additive/CauchyDavenport.lean

```diff
@@ -176,17 +176,20 @@ lemma cauchy_davenport_minOrder_mul (hs : s.Nonempty) (ht : t.Nonempty) :
       (WithTop.coe_le_coe.2 aux2).trans' fun h ↦
         hstg.le.trans <| h.trans <| add_le_add_right aux2 _
 
+end General
+
 /-- The **Cauchy-Davenport Theorem** for torsion-free groups. The size of `s * t` is lower-bounded
 by `|s| + |t| - 1`. -/
 @[to_additive
 "The **Cauchy-Davenport theorem** for torsion-free groups. The size of `s + t` is lower-bounded
 by `|s| + |t| - 1`."]
-lemma cauchy_davenport_mul_of_isTorsionFree (h : IsTorsionFree α)
-    (hs : s.Nonempty) (ht : t.Nonempty) : #s + #t - 1 ≤ #(s * t) := by
-  simpa only [h.minOrder, min_eq_right, le_top, Nat.cast_le]
+lemma cauchy_davenport_of_isMulTorsionFree [DecidableEq G] [CommGroup G] [IsMulTorsionFree G]
```

Comment:
As you point out, this change is weird and unfortunate... Is it caused by `minOrder_eq_top` not being general enough, or what's going on?

Reply:
Oh god, yes, big brainfart